madsim-tokio/src/sim/runtime.rs

```diff
@@ -188,4 +191,200 @@ mod tests {
+            let join_handle = rt.spawn(async move {
+                Handle::current().spawn(async {
+                    Handle::current().spawn(async {
+                        Handle::current().spawn(async {
+                            Handle::current().spawn(async {
+                                Handle::current().spawn(async {
+                                    Handle::current().spawn(async {
+                                        Handle::current().spawn(async {
+                                            Handle::current().spawn(async {
+                                            });
+                                        });
+                                    });
+                                });
+                            });
+                        });
+                    });
+                });
+            });
+            join_handle.await.unwrap();
+        });
+    }
+
+    #[test]
+    fn runtime_spawn_shallow_nested() {
+        let runtime = madsim::runtime::Runtime::new();
+        runtime.block_on(async move {
+            let rt = Runtime::new().unwrap();
+            let join_handle = rt.spawn(async move {
+                Handle::current().spawn(async {});
+            });
+            join_handle.await.unwrap();
+        });
+    }
+
+    #[test]
+    fn runtime_spawn_no_nesting() {
+        let runtime = madsim::runtime::Runtime::new();
+        runtime.block_on(async move {
+            let rt = Runtime::new().unwrap();
+            Handle::current().spawn(async {});
+        });
+    }
+
+    #[test]
+    fn runtime_spawn_empty() {
+        let runtime = madsim::runtime::Runtime::new();
+        runtime.block_on(async move {
+            let rt = Runtime::new().unwrap();
+            let join_handle = rt.spawn(async move {
+            });
+            join_handle.await.unwrap();
+        });
+    }
+
+    #[test]
+    fn runtime_spawn_no_tokio_rt() {
+        init_logger();
+        let runtime = madsim::runtime::Runtime::new();
+        runtime.block_on(async move {
+            let rt = Runtime::new().unwrap();
+            let join_handle = Handle::current().spawn(async move {
+            });
+            join_handle.await.unwrap();
+            tracing::info!("finished");
+        });
+    }
+
+
+    #[test]
+    fn runtime_spawn_indirect() {
+        init_logger();
+        let runtime = madsim::runtime::Runtime::new();
+        runtime.block_on(async move {
+            let rt = Runtime::new().unwrap();
+            let join_handle = (&Handle).spawn(async move {
+            });
+            join_handle.await.unwrap();
+            tracing::info!("finished");
+        });
+    }
+
+    #[test]
+    fn runtime_spawn_direct() {
+        init_logger();
+        let runtime = madsim::runtime::Runtime::new();
+        runtime.block_on(async move {
+            let join_handle = madsim::task::spawn(async {});
+            join_handle.await.unwrap();
+        });
```

Comment:
The first test out of these 3 highlighted fails (`runtime_spawn_no_tokio_rt`).

That's really strange because `Handle::current()`'s definition is just:

```
fn current() {
  Handle
}
```

So if everything is inlined, it just calls `madsim::task::spawn`.
Somehow this future executes differently..

Reply:
This test uses tokio handle rather than madsim tokio. It is wrong.